src/Apps/Preferences/__tests__/PreferencesApp.jest.tsx

```diff
@@ -1,145 +1,261 @@
-    let subscribeToAllCheckbox = checkboxes[0]
-    let unsubscribeFromAllCheckbox = checkboxes.pop()
-
-    fireEvent.click(checkboxes[3])
-
-    expect(subscribeToAllCheckbox).not.toBeChecked()
-    expect(unsubscribeFromAllCheckbox).not.toBeChecked()
+    expect(screen.getByLabelText("Recommended for You")).not.toBeChecked()
+    expect(screen.getByLabelText("Editorial")).not.toBeChecked()
+    expect(screen.getByLabelText("Guidance on Collecting")).not.toBeChecked()
+    expect(screen.getByLabelText("Custom Alerts")).not.toBeChecked()
+    expect(screen.getByLabelText("Offers on Saved Artworks")).not.toBeChecked()
+    expect(screen.getByLabelText("Product Updates")).not.toBeChecked()
   })
 
-  it.skip("allows user to check all boxes with subscribe to all", () => {
-    render(
-      <HeadProvider>
-        <PreferencesApp></PreferencesApp>
-      </HeadProvider>
-    )
-
-    expect(screen.getByText("Unsubscribe from all")).toBeInTheDocument()
-
-    let checkboxes = screen.getAllByRole("checkbox")
-    let subscribeToAllCheckbox = checkboxes[0]
-    let unsubscribeFromAllCheckbox = checkboxes.pop()
-
-    fireEvent.click(subscribeToAllCheckbox)
-
-    expect(subscribeToAllCheckbox).toBeChecked()
+  it("saves preferences correctly on form submission", async () => {
+    const submitMutationMock = jest.fn(() => Promise.resolve())
+    ;(useEditNotificationPreferences as jest.Mock).mockReturnValue({
+      submitMutation: submitMutationMock,
+    })
 
-    expect(unsubscribeFromAllCheckbox).not.toBeChecked()
+    renderWithRelay({
+      Viewer: () => ({
+        notificationPreferences: [
+          { name: "recommendedByArtsy", status: "SUBSCRIBED" },
+          { name: "artWorldInsights", status: "SUBSCRIBED" },
+        ],
+      }),
+    })
 
-    checkboxes.forEach(checkbox => {
-      expect(checkbox).toBeChecked()
+    fireEvent.click(screen.getByLabelText("Recommended for You"))
+    fireEvent.click(screen.getByLabelText("Art World Insights"))
+    fireEvent.click(screen.getByText("Save"))
+
+    await waitFor(() => {
+      expect(submitMutationMock).toHaveBeenCalledWith({
+        input: {
+          subscriptionGroups: [
+            { name: "recommendedByArtsy", status: "UNSUBSCRIBED" },
+            { name: "artWorldInsights", status: "UNSUBSCRIBED" },
+          ],
+        },
+      })
     })
+  })
 
-    fireEvent.click(subscribeToAllCheckbox)
+  describe("toasts", () => {
+    it("displays a success toast on successful form submission", async () => {
+      const submitMutationMock = jest.fn(() => Promise.resolve())
+      ;(useEditNotificationPreferences as jest.Mock).mockReturnValue({
+        submitMutation: submitMutationMock,
+      })
+
+      renderWithRelay({
+        Viewer: () => ({
+          notificationPreferences: [],
+        }),
+      })
+
+      fireEvent.click(screen.getByText("Save"))
+      const { sendToast } = require("@artsy/palette").useToasts()
```

Comment:
Since we're already mocking this module, we can do something like this at the top of the file (in the describe block):

```
const mockUseToasts = useToasts as jest.Mock
```

And then in here, can mock the implementation:

```tsx
const spy = jest.fn()

mockUseToasts.mockReturnValue({
  sendToast: spy
})

...

expect(spy).toHaveBeenCalled(...)
```

And then we don't need the eslint disable rule at the top, as we'll be using that import.